// Excecao 1

public class Excecao1 extends Exception {
   public Excecao1() { }
   public Excecao1(String msg) {
       super(msg);
   }
}

//Teste 1

public class Teste1 {

    public static int divide(int a,int b) {
        return a/b;
    }
    public static int f(int a,int b) {
        return divide(a,b);
    }
    public static void main(String args []) {
        System.out.println(f(5,2));
        System.out.println(f(3,0));
        System.out.println(f(4,1));
    }
}

//Teste 2

public class Teste2 {

    public static int divide(int a,int b) {
       // if (b==0) throw new Excecao1("Divisao por zero!"); ERRO de compilação
        return a/b;
    }
    public static int f(int a,int b) {
        return divide(a,b);
    }
    public static void main(String args []) {
        System.out.println(f(5,2));
        System.out.println(f(3,0));
        System.out.println(f(4,1));
    }

}


//Teste 3

public class Teste3 {
     public static int divide(int a,int b) throws Excecao1{
      if (b==0) throw new Excecao1("Divisao por zero!");
        return a/b;
    }
//    public static int f(int a,int b) {
//       // return divide(a,b); //ERRO de compilação
//    }
    public static void main(String args []) {
//        System.out.println(f(5,2));
//        System.out.println(f(3,0));
//        System.out.println(f(4,1));
    }
}


//Teste 4

public class Teste4 {
     public static int divide(int a,int b) throws Excecao1{
      if (b==0) throw new Excecao1("Divisao por zero!");
        return a/b;
    }
    public static int f(int a,int b) throws Excecao1 {
        return divide(a,b); //ERRO de compilação
    }
    public static void main(String args []) {
//        System.out.println(f(5,2)); // Erro de complicação
//        System.out.println(f(3,0));
//        System.out.println(f(4,1));
    }
}


//Teste 5

public class Teste5 {
    public static int divide(int a,int b) throws Excecao1{
      if (b==0) throw new Excecao1("Divisao por zero!");
        return a/b;
    }
    public static int f(int a,int b) throws Excecao1 {
        return divide(a,b); //ERRO de compilação
    }
    public static void main(String args [])throws Excecao1 {
        System.out.println(f(5,2));
        System.out.println(f(3,0));
        System.out.println(f(4,1));
    }
}


//Teste 6


public class Teste6 {
    public static int divide(int a,int b) {
       try {
           if (b==0) throw new Excecao1("Divisao por zero!");
           return a/b;
        } catch (Exception e) {
            e.printStackTrace();
            System.out.println("Resultado nao valido:");
            return 0;
        }
    }

    public static int f(int a,int b)  {
        return divide(a,b);
    }
    public static void main(String args []){
        System.out.println(f(5,2));
        System.out.println(f(3,0));
        System.out.println(f(4,1));
    }
}


